pkg/controller/argocd/policyrule.go

```diff
@@ -8,6 +8,19 @@ import (
 func policyRuleForApplicationController() []v1.PolicyRule {
 
 	return []v1.PolicyRule{
+		{
+			APIGroups: []string{
+				"*",
```

Comment:
Can you explain why this is added?

Reply:
These permissions were added as part of the ClusterRole (https://github.com/argoproj-labs/argocd-operator/pull/314/files#diff-e7925228b00ce545bae87b56e0167a16b69ae670b1ab4fda380089c88b766900L180), but now we don't have a cluster role associated with the service account of the application controller. 
So i have moved these permissions to the Role associated with the service account. This provides the application controller to read all the resources deployed within the same namespace.

Reply:
@chetan-rns Do we need to move the following here?
```
{
			NonResourceURLs: []string{
				"*",
			},
			Verbs: []string{
				"get",
				"list",
			},
		},
    ```

Reply:
`NonResourceURL` rules can only be added to a ClusterRole, since they are cluster scoped. So we cannot add them to a Role.